include/zephyr/linker/common-rom/common-rom-kernel-device-apis.ld

```diff
@@ -0,0 +1,7 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```

Comment:
that might become overwhelming as we add more APIs, realistically, we will have 100+ APIs there at some point, seems like it should be in its own file

Reply:
Fair point, but then include it in zephyr/linker/common-rom/common-rom-kernel-devices.ld and not in include/zephyr/linker/common-rom.ld so you won't have to fix the esp's that do not include common-rom.ld, and anyway it will be more logical there.

Reply:
I moved the section to zephyr/linker/common-rom/common-rom-kernel-devices.ld and restored the esp linker files :)